lib/web_ui/lib/src/engine/window.dart

```diff
@@ -214,17 +217,30 @@ class EngineWindow extends ui.Window {
     scheduleFrameCallback!();
   }
 
+  @visibleForTesting
+  JsUrlStrategy? urlStrategy = ui.debugEmulateFlutterTesterEnvironment
+      ? null
+      : convertToJsUrlStrategy(const HashUrlStrategy());
```

Comment:
This will convert the object on every access, which can be fixed by removing the need for conversion (see other comment). Then this getter could be simplified to something like:

```dart
UrlStrategy? get urlStrategy => ui.debugEmulateFlutterTesterEnvironment
      ? null
      : _customUrlStrategy ?? const HashUrlStrategy();
```

Reply:
This isn't a getter. The conversion is done only once.